src/test/java/org/openmrs/contrib/qaframework/automation/ClinicalVisitSteps.java

```diff
@@ -288,7 +287,7 @@ public void endPatientVisit() {
 
 	@Then("the system ends the patient visit")
 	public void systemEndsPatientVisit() {
-		assertNull(visitsDashboardPage.getActiveVisit());
```

Comment:
Asserting page seems not to achieve the intended goal for the [step](https://github.com/openmrs/openmrs-contrib-qaframework/blob/551f9b36d0437f3c6f4466b74a971a2ad91842ea/src/test/resources/features/refapp-2.x/14-clinicalVisit/clinicalVisit.feature#L46). The element consumed within visitDashboardPage.getActiveVisit() seems to get stale and makes the ended visit to take longer to be inactive and thus makes the test fail because it finds the visit still [active](https://github.com/openmrs/openmrs-contrib-qaframework/blob/551f9b36d0437f3c6f4466b74a971a2ad91842ea/src/test/java/org/openmrs/contrib/qaframework/page/PatientVisitsDashboardPage.java#L83)

Reply:
> seems to gets stale when invoked and makes the ended visit to take longer to be inactive and thus makes the test fail because it finds the visit still [active](https://github.com/openmrs/openmrs-contrib-qaframework/blob/551f9b36d0437f3c6f4466b74a971a2ad91842ea/src/test/java/org/openmrs/contrib/qaframework/page/PatientVisitsDashboardPage.java#L83)

And what is the problem with these changes, Are you trying to mean, there are active visits when already the system ends the visit ??

Reply:
> And what is the problem with these changes, Are you trying to mean, there are active visits when already the system ends the visit ??, its simply failing because there are no active visits.

@sherrif10 it is actually failing because when the visit is ended, it appears to still be active at the time the page loads so the assertNull is meant to pass if the there are no active visits, but the fact that there's a lag, the visit still shows that it is active which then makes the test to fail on the assertNull.

Reply:
Asserting Null was catered because by default if there are no active visit, the system is supposed to return null , it was implemented it here https://github.com/openmrs/openmrs-contrib-qaframework/blob/master/src/test/java/org/openmrs/contrib/qaframework/page/PatientVisitsDashboardPage.java#L83.  there are million ways of fixing this, One way of fixing this is get rid of null and simply check the existance of any element on the field or instead of returning null, return anything like a page or a field on the page.

Reply:
> And what is the problem with these changes

> Asserting the page seems not to achieve the intended goal for the [step](https://github.com/openmrs/openmrs-contrib-qaframework/blob/551f9b36d0437f3c6f4466b74a971a2ad91842ea/src/test/resources/features/refapp-2.x/14-clinicalVisit/clinicalVisit.feature#L46). 

have you had a look at what [getVisitList()](https://github.com/openmrs/openmrs-contrib-qaframework/blob/551f9b36d0437f3c6f4466b74a971a2ad91842ea/src/test/java/org/openmrs/contrib/qaframework/page/PatientVisitsDashboardPage.java#L90)can achieve?

Reply:
> > And what is the problem with these changes
> 
> > Asserting the page seems not to achieve the intended goal for the [step](https://github.com/openmrs/openmrs-contrib-qaframework/blob/551f9b36d0437f3c6f4466b74a971a2ad91842ea/src/test/resources/features/refapp-2.x/14-clinicalVisit/clinicalVisit.feature#L46).
> 
> have you had a look at what [getVisitList()](https://github.com/openmrs/openmrs-contrib-qaframework/blob/551f9b36d0437f3c6f4466b74a971a2ad91842ea/src/test/java/org/openmrs/contrib/qaframework/page/PatientVisitsDashboardPage.java#L90-L92) can achieve?

@kdaud  won't this just return all the visits, both active and ended visits? Meaning it may not help us to be sure if the visit is ended or not, since it will return all visits a patient has ever made, whether active or not. Any thoughts?